pkg/apiserver/rest/spy_client.go

```diff
@@ -129,10 +130,15 @@ func (c *spyLegacyStorageClient) Get(ctx context.Context, name string, options *
 	return nil, nil
 }
 
+func (c *spyLegacyStorageClient) NewList() runtime.Object {
+	// stub for now so that spyLegacyStorageClient implements rest.Lister
+	return nil
+}
+
 func (c *spyLegacyStorageClient) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
 	c.spy.record("LegacyStorage.List")
 	klog.Info("method: LegacyStorage.List")
-	return nil, nil
+	return &dummyObject{}, nil
```

Comment:
I'm not sure why do we need to return a dummyObject here?

Reply:
The mode 2 List test was failing with:

```
Received unexpected error:
<nil> is not a list: expected pointer, but got invalid kind
```